pitch.md

```diff
@@ -1,69 +1,172 @@
+Furthermore, the nastiness of C++'s templates are not just inherited from C++'s
+general style. In Scala, similar things start to happen when you try to implement
+dependent types.
+
+Ultimately, the real problem is that programmers only have incidental control
+over compile-time behavior. I want to design a language where there is
+direct and encouraged access to compile-time behavior. This is extremely ambitious,
+so there are many limits that will be discussed later.
 
 ## The user and a language
-This section describes who the project would serve and why a language might be a
-good way to meet their needs.
 
 
 ### What's the need?
-_What need is met by your idea? Who are you helping? What is that person's
-experience like now? What would their experience be like if you could help 
-them?_
 
+The big need I see is that with current languages, extensibility ends at
+compile-time features. That is not completely true: again, it is _possible_
+to develop a lot of functionality by leveraging what already exists,
+especially when you get into the realm of macros.  
+And Haskell encourages modifications to the compiler, such that something
+like Idris (which has dependent types) can be built off of it, which is also
+pretty successful. Even so, these are not particularly friendly.
+I want to explore the possibility of extensibility in this region as well.
+
+There is also a second need.
+Supposedly, this would be a general-purpose programming language, whose compile-time features
+were especially easy to extend. That could mean that it would be extremely useful
+for people writing code that needs exotic compile-time features, such as verification:
+but, in actuality, if it were to be practically useful, it would probably be as a playground
+for working with different variations on compile-time concepts such as dependent types.
+So it is also potentially useful as a programming language researcher's aide.
 
 ### Why a language?
-_Why is a DSL appropriate for your user(s)? How does it address the need?_
+
+Well, ultimately, this is a language about features of programming languages.
+It needs to be in close reflection to programming languages, so this seems to be a no-brainer.
 
 
 ### Why you?
-_What excites you about this idea? How did you come up with it?_
+
+What first interested me about this kind of idea was learning about D. It's essentially
+C++ done right. D offers much closer control over compile-time behavior. In particular,
+D has such features as compile-time "if" statements, compile-time arguments to functions,
+and a much better system than C++'s templates.
+
+The other thing is that I really want to explore type systems.
+First, building a language like this would force me to look at them.
+But also, _having_ a language like this, if it was actually sufficiently expressive,
+would be helpful in learning-through-implementing those ideas.
 
```

Comment:
Same question that I asked about dependent types, would it be possible to extend an existing language to have a type system? Maybe using a constraint solver?